File: app/src/main/java/com/juawapps/newsspread/di/AppModuleApi.java
package com.juawapps.newsspread.di;

import com.juawapps.newsspread.data.api.ApiProvider;
import com.juawapps.newsspread.data.api.NewsapiService;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Database module
 */
@Module
public class AppModuleApi {

    @Provides
    @Singleton
    NewsapiService provideNewsApiService() {
        return ApiProvider.getApiService();
    }

}

File: app/src/main/java/com/juawapps/newsspread/di/base/AppModuleApi.kt
package com.juawapps.newsspread.di.base

import com.juawapps.newsspread.data.api.ApiProvider
import com.juawapps.newsspread.data.api.NewsapiService

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

/**
 * Database module
 */
@Module
class AppModuleApi {

    @Provides
    @Singleton
    internal fun provideNewsApiService(): NewsapiService {
        return ApiProvider.apiService
    }

}
